[pvr] Fix channel scan seg faults, init seg faults and code cleanup

AlwinEsch opened the pull request:
The following changes fix segmention faults during open of PVR channel scan dialog.

FernetMenta commented on xbmc/pvr/PVRManager.cpp:
```diff
@@ -600,7 +603,6 @@ bool CPVRManager::SetWakeupCommand(void)
 
 bool CPVRManager::StartUpdateThreads(void)
 {
-  StopUpdateThreads();
```
please elaborate on this. why does stopping a thread before starting it fix a segfault? a stopped thread is a pre-condition for starting it.

AlwinEsch replied:
Has added it back, during the last calls it was OK with the rest.

AlwinEsch commented:
Is something wrong today on github? Also it don't show me there the last update.

AlwinEsch committed:
[pvr] Fix freezes and crashes with started channel scan

Conflicts:
	xbmc/pvr/addons/PVRClients.cpp

AlwinEsch commented:
OK for you to do a UUID for add-on together with https://github.com/xbmc/xbmc/pull/6877 or on a separate request inside KODI?

AlwinEsch committed:
[pvr] Fix possible system frezzes during manager thread stop

AlwinEsch committed:
[pvr] Remove not needed trailing spaces

Conflicts:
	xbmc/epg/EpgContainer.cpp
	xbmc/pvr/PVRActionListener.cpp
	xbmc/pvr/PVRActionListener.h
	xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp
	xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.h
	xbmc/pvr/timers/PVRTimerInfoTag.cpp
	xbmc/pvr/windows/GUIWindowPVRBase.cpp
	xbmc/pvr/windows/GUIWindowPVRGuide.cpp

AlwinEsch commented:
Request is fully updated and contains now several changes and fixes.

1.: Fix and possible system freeze on pvr restart due to still required thread work and was stopped.
2.: Fix freeze and seq fault with opened and closed channel scan
3.: Fix the PVR database about faults with already present channels and asked id = -1
4.: Remove a not needed id check, this is always done before
5.: Fix a sometimes produced seq faults during pvr start, which was produced by two threads which has calls sometimes the same EPG database access together.
6.: Remove from pvr system the trailing spaces (don't want to look all the time about it ;-) )
7.: Sort the epg header files alphabetical
8.: Swap over everything there to std::
9.: Use the already defined EPGMAP everywhere
10. Sort also the pvr files alphabetical, also several double headers are removed.

@FernetMenta @opdenkamp can you do a review.
The database part is now really better.

FernetMenta commented:
looks sane to me.

xhaggi commented:
@opdenkamp mind taking a look too.

AlwinEsch committed:
[epg] Break update loop if PVRManager is disabled

fritsch commented:
jenkins build this please

AlwinEsch commented:
Hello @fritsch, there is somewhere a epg database error inside which increase the size of it on every startup, searching currently for it.

AlwinEsch commented:
Has found the fault, has removed a part which was still required, the related changes was currently not present here but one half of them is the base of this request.

AlwinEsch commented:
So, everything needed is inside and the bad part is removed.
Allowed to kill me ;-) -->
jenkins build this please

fritsch commented:
@AlwinEsch Lol - the "shooting" needs ack of the team - and yeah: -1 to that :-)

AlwinEsch commented:
What the status about this? Merge in?

AlwinEsch commented:
@opdenkamp has changed to CSingleLock on your node, thanks.

fritsch commented:
@AlwinEsch this needs a rebase - but we still want it right?

AlwinEsch commented:
How to handle this at the moment? Ready to merge.
Sorry for the late update, since last days I have not so much time due to private reasons.

AlwinEsch commented:
With this changes or also without?

stefansaraev commented:
with this PR, of course. it crashes shortly after I try enter client specific. without this pr - all fine. I can ping you in irc tonight with a propper debug crashlog if you cant reproduce.

AlwinEsch commented:
Thanks, here it is OK, hope with your crashlog I find the source.

stefansaraev commented:
ok, it'sa bit more complicated. so:

this PR alone - no crashes. testing with pvr.vdr.vnsi 3a28e39 (without your channelscanner patch). vdr is 2.0.0. vdr-plugin-vnsiserver is 36e2b61 (before your newscanner commits)

this PR + vdr-plugin-vnsiserver-1772c92 (your channelsscaner included) + https://github.com/kodi-pvr/pvr.vdr.vnsi/pull/5 on top of pvr.vdr.vnsi-3a28e39 = crash in tv -> client options, and tv -> general -> search for channels.

call me stupid, but I am lost. not sure if it's this PR or https://github.com/kodi-pvr/pvr.vdr.vnsi/pull/5 is the culprit. 

crashlogs: 
http://sprunge.us/iNML (client options) 
http://sprunge.us/jJDa (channel scanner)

MartijnKaijser commented:
jenkins build this please
is this something we can shove in 15.0? Seems like a bunch of fixes and needs to go to the beta testers.

MartijnKaijser commented:
So what's the story n thus PR? Split into smaller parts or this good for merge after another review?

opdenkamp commented:
fixes should be split up and included in 15, rest should go into a separate pr and pulled later (the cosmetics and increment changes)

MartijnKaijser commented:
Let me close it for now. New PR can be opened if wanted/needed and is rebased on current master branch.
